can/can.cpp

```diff
@@ -0,0 +1,255 @@
+#ifndef CONFIG_DISABLE_MODULE_CAN
+
+#include "can.hpp"
+
+namespace halex {
+
+Can::Can(CAN_HandleTypeDef *hcan) : hcan(hcan) {
+
+}
+
+CAN_StatusType Can::init() {
+    if (HAL_CAN_Init(hcan) != HAL_OK) {
+        return CAN_StatusType::CAN_Fail_Init;
+    }
+    return static_cast<CAN_StatusType>(HAL_CAN_Start(hcan));
+}
+
+void Can::setFilterActivationState(uint32_t state) {
+    filterConfig.FilterActivation = state;
+}
+
+void Can::setFilterMode(CAN_FilterMode filterMode) {
+    this->filterMode = filterMode;
+
+    switch (this->filterMode) {
+        case CAN_FilterMode::PATH_ONE_TYPE_STD_OR_EXT_ID_GROUP:
+            filterConfig.FilterMode = CAN_FILTERMODE_IDMASK;
+            filterConfig.FilterScale = CAN_FILTERSCALE_32BIT;
+            break;
+
+        case CAN_FilterMode::PATH_TWO_TYPE_STD_OR_EXT_ID:
+            filterConfig.FilterMode = CAN_FILTERMODE_IDLIST;
+            filterConfig.FilterScale = CAN_FILTERSCALE_32BIT;
+            break;
+
+        case CAN_FilterMode::PATH_TWO_TYPE_STD_ID_GROUP:
+            filterConfig.FilterMode = CAN_FILTERMODE_IDMASK;
+            filterConfig.FilterScale = CAN_FILTERSCALE_16BIT;
+            break;
+
+        case CAN_FilterMode::PATH_FOUR_TYPE_STD_ID:
+
+            filterConfig.FilterMode = CAN_FILTERMODE_IDLIST;
+            filterConfig.FilterScale = CAN_FILTERSCALE_16BIT;
+            break;
+
+        case CAN_FilterMode::PATH_ALL_ID:
+            filterConfig.FilterMode = CAN_FILTERMODE_IDMASK;
+            filterConfig.FilterScale = CAN_FILTERSCALE_32BIT;
+            filterConfig.FilterIdHigh = 0x0000;
+            filterConfig.FilterIdLow = 0x0000;
+            filterConfig.FilterMaskIdHigh = 0x0000;
+            filterConfig.FilterMaskIdLow = 0x0000;
+            break;
+    }
+}
+
+void Can::setFilterBank(uint32_t CAN2_filterBankNumber) {
+    filterConfig.FilterBank = 0;
+    filterConfig.SlaveStartFilterBank = CAN2_filterBankNumber;
+}
+
+void Can::setStoreRxFifo(uint32_t rxFifo) {
+    filterConfig.FilterFIFOAssignment = rxFifo;
+}
```

Comment:
```suggestion
void Can::setFilterFIFOAssignment(uint32_t value) {
    filterConfig.FilterFIFOAssignment = value;
}
```

[WANT] filterConfig の名前に合わせたいです

Reply:
cf3ca2fcc8c8b34f3ec8f71cdc46f48ae8cada48 で対応